OpenUpSA/wazimap-ng
# Tests for All details api enpoint and versions

## Description
Added tests for Geo versions all_details:

Test Cases:

* **Highlights**:Test for highlights data that come in all details api

  *  **test_highlight_without_data** : If we do not have any highlights for profile
  * **test_highlight_with_valid_data**: Just check the output for highlights where we have valid data (Happy path - Test with optimal data) 
  * **test_multiversion_highlight_with_valid_data**: Test highlights if there are multiple versions for a profilecreate multiple versions with valid data and set one version as default. 
   Check if appropriate data is returned according to the version passed to all_details endpoint
    * Check  if we get highlights for the default version(version1) when we do not pass the version in API
    * Check if we get highlights for the version we are passing in the ApiWe get highlights_v1 for version1 and highlights_v2 for version2.

  - **test_highlight_sibling_with_multiversion** : Test highlight for denominator type sibling and output should be different in case a sibling exists for a specific version.create multiple versions, create a sibling that exists for a specific version, create data for  that geography
    - Test data returned for version1 
    - Test if data returned for version2 is not the same as data returned for version1 (It tests that data from newly added sibling affected output for version2)
    
    For both versions, we will get the same highlight (highlight_v1) but the return value should be changed according to version and siblings included in calculations.

- **Profile Key Metrics**
 
   - **test_multiversion_metrics_data** : Test ky metrics if there are multiple versions for a profilecreate multiple versions with valid data and set one version as default.Check if the appropriate metric is returned according to the version passed to the all_details endpoint

     - Check  if we get metrics for the default version when we do not pass the version in API (metric_v1 is returned)

     - Check if we get metrics for the version we are passing in the ApiWe get metric_v1 for version1 and metric_v2 for version2.

  - **test_metric_sibling_with_multiversion** : Test metrics for denominator type sibling and output should be different in case a sibling exists for a specific version.

       create multiple versions, create a sibling that exists for a specific version, create data for  that geography

       - Test data returned for version1 

       - Test if data returned for version2 is not the same as data returned for version1 (It tests that data from newly added sibling affected output for version2)

       For both versions, we will get the same highlight (metric_v1) but the return value should be changed according to version and siblings included in calculations.

- **Parents data in all detailts (TestParentLayersData)**:  tests when there are two versions available for the same geo but one of its ancestor-siblings only exists in the selected version.

 - **Test for Child indicator data ( TestChildrenData)** : when there are two versions available for the same geo but one of its children only exist in the selected version


## Related Issue
https://wazimap.atlassian.net/browse/WNCM-122
https://wazimap.atlassian.net/browse/WNCM-99

## How to test it locally

## Changelog

- Moved tests for general views to separate folder and files
- Updated test_metrics_serializer.py to fix breaking tests
- Updated metrics_serializer.py to get version for denominator option


### Added

### Updated

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
